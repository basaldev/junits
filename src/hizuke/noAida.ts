export default function noAida(date1: Date, date2: Date, date3: Date): boolean {
  return date1.getTime() >= date2.getTime() && date1.getTime() <= date3.getTime();
}
