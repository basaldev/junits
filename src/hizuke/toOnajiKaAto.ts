export default function toOnajiKaAto(date1: Date, date2: Date): boolean {
  return date1.getTime() >= date2.getTime();
}
