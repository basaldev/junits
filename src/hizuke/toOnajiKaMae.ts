export default function toOnajiKaMae(date1: Date, date2: Date): boolean {
  return date1.getTime() <= date2.getTime();
}
