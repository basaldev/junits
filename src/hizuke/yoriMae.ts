export default function yoriMae(date1: Date, date2: Date): boolean {
  return date1.getTime() < date2.getTime();
}
