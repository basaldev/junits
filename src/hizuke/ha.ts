import converToDate, { Datetime } from './convertToDate';
import yoriMae from './yoriMae';
import yoriAto from './yoriAto';
import toOnaji from './toOnaji';
import toOnajiKaMae from './toOnajiKaMae';
import toOnajiKaAto from './toOnajiKaAto';
import noAida from './noAida';

export default function ha(date1: Date, date2: Date) {
  return {
    yoriMae: () => yoriMae(date1, date2),
    yoriAto: () => yoriAto(date1, date2),
    toOnaji: () => toOnaji(date1, date2),
    toOnajiKaMae: () => toOnajiKaMae(date1, date2),
    toOnajiKaAto: () => toOnajiKaAto(date1, date2),
    to: (date3: Datetime) => ({ noAida: () => noAida(date1, date2, converToDate(date3)) })
  };
}
