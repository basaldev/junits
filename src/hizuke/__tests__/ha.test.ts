jest.mock('../yoriMae');
jest.mock('../yoriAto');
jest.mock('../toOnaji');
jest.mock('../toOnajiKaMae');
jest.mock('../toOnajiKaAto');
jest.mock('../noAida');

import yoriMae from '../yoriMae';
import yoriAto from '../yoriAto';
import toOnaji from '../toOnaji';
import toOnajiKaMae from '../toOnajiKaMae';
import toOnajiKaAto from '../toOnajiKaAto';
import noAida from '../noAida';
import ha from '../ha';

describe('ha', () => {
  let date1;
  let date2;
  let date3;
  let expectedHaReturn;

  beforeEach(() => {
    date1 = new Date(2019, 4, 7, 13, 16, 20);
    date2 = new Date(2020, 4, 7, 13, 16, 20);
    date3 = new Date(2021, 4, 7, 13, 16, 20);
    expectedHaReturn = {
      to: expect.any(Function),
      toOnaji: expect.any(Function),
      toOnajiKaAto: expect.any(Function),
      toOnajiKaMae: expect.any(Function),
      yoriAto: expect.any(Function),
      yoriMae: expect.any(Function)
    };
  });

  it('should return proper members', () => {
    expect(ha(date1, date2)).toMatchSnapshot(expectedHaReturn);
  });

  it('should call yoriMae', () => {
    ha(date1, date2).yoriMae();
    expect(yoriMae).toHaveBeenLastCalledWith(date1, date2);
  });

  it('should call yoriAto', () => {
    ha(date1, date2).yoriAto();
    expect(yoriAto).toHaveBeenLastCalledWith(date1, date2);
  });

  it('should call toOnaji', () => {
    ha(date1, date2).toOnaji();
    expect(toOnaji).toHaveBeenLastCalledWith(date1, date2);
  });

  it('should call toOnajiKaMae', () => {
    ha(date1, date2).toOnajiKaMae();
    expect(toOnajiKaMae).toHaveBeenLastCalledWith(date1, date2);
  });

  it('should call toOnajiKaAto', () => {
    ha(date1, date2).toOnajiKaAto();
    expect(toOnajiKaAto).toHaveBeenLastCalledWith(date1, date2);
  });

  it('should call to().noAida', () => {
    ha(date1, date2)
      .to(date3)
      .noAida();
    expect(noAida).toHaveBeenLastCalledWith(date1, date2, date3);
  });
});
