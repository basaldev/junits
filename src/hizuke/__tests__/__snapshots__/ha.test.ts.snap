// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ha should return proper members 1`] = `
Object {
  "to": Any<Function>,
  "toOnaji": Any<Function>,
  "toOnajiKaAto": Any<Function>,
  "toOnajiKaMae": Any<Function>,
  "yoriAto": Any<Function>,
  "yoriMae": Any<Function>,
}
`;
